### PR TITLE
AF v2 - sample-search artifact argument addition

### PR DESCRIPTION
**Pull request by orhovy**
## Status
- [x] In Progress

## Related Issues
fixes: https://github.com/demisto/etc/issues/23139

## Description
added a new argument to search-samples command.
changed context and human readable of samples-result command. now every entry in table will be set in a new entry in war room , and for each entry its artifacts will be added.

## Minimum version of Demisto
- [x] 4.5.0

## Does it break backward compatibility?
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py [hunk 1]
```diff
@@ -295,40 +297,44 @@ def http_request(url_suffix, method='POST', data={}, err_operation=None):
     return parse_response(res, err_operation)
 
 
-def validate_sort_and_order(sort, order):
-    if sort and not order:
+def validate_sort_and_order_and_artifact(sort, order, artifact_source):
+    if artifact_source == 'true' and sort:
+        return_error('artifact and sort are not supported for scan and scroll together, Please Remove one.')
```
Comment (bakatzir) [inline Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py:302]: what is scan and scroll?

File: Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py [hunk 2]
```diff
@@ -1145,23 +1153,43 @@ def samples_search_results_command():
...
+                hr = samples_search_result_hr(result, status)
+                return_outputs(readable_output=hr, outputs=context, raw_response=results)
+            else:
+                hr = tableToMarkdown(f'Search Samples Result is {status}', result)
+                hr += tableToMarkdown(f'Artifacts for Sample: ', [])
+                return_outputs(readable_output=hr, outputs=context, raw_response=results)
+
+
+def samples_search_result_hr(result, status):
+    Artifact = result.pop('Artifact')
```
Comment (bakatzir) [inline Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py:1180]: why the name var is capital?

File: Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.yml [hunk 3]
```diff
@@ -134,10 +134,22 @@ script:
       name: last_updated
       required: false
       secret: false
+    - auto: PREDEFINED
+      default: true
+      defaultValue: 'true'
+      description: 'Set True when artifacts of samples are wanted '
```
Comment (bakatzir) [inline Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.yml:140]:
```suggestion
      description: 'Wether artifacts of samples should be returned.'
```

File: Packs/AutoFocus/Integrations/AutofocusV2/CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,6 @@
 ## [Unreleased]
--
+Added to ***autofocus-search-samples*** command a new argument, artifact which is set true by default
+Added to ***autofocus-samples-search-results*** artifacts results to context and divided human readable to a new entry for each sample.
```
Comment (bakatzir) [inline Packs/AutoFocus/Integrations/AutofocusV2/CHANGELOG.md:3]:
```suggestion
Added to ***autofocus-search-samples*** command the *artifact* argument which is set true by default and retrieves the artifacts of the sample.
```

File: Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py [hunk 5]
```diff
@@ -297,17 +299,45 @@ def http_request(url_suffix, method='POST', data={}, err_operation=None):
     return parse_response(res, err_operation)
 
 
-def validate_sort_and_order_and_artifact(sort, order, artifact_source):
+def validate_sort_and_order_and_artifact(sort: Optional[str], order: Optional[str], artifact_source: Optional[str])\
+        -> Optional[str]:
+    """
+    Function that validates the arguments combination.
+    sort and order arguments have come together.
```
Comment (bakatzir) [inline Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py:306]:
```suggestion
    sort and order arguments must be defined together.
```

File: Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py [hunk 6]
```diff
@@ -297,17 +299,45 @@ def http_request(url_suffix, method='POST', data={}, err_operation=None):
...
+    Sort and order cant appear with artifact.
```
Comment (bakatzir) [inline Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py:307]:
```suggestion
    Sort and order can't appear with artifact.
```

File: Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py [hunk 7]
```diff
@@ -297,17 +299,45 @@ def http_request(url_suffix, method='POST', data={}, err_operation=None):
...
+        true if arguments are valid for the request, false otherwise.
+    """
     if artifact_source == 'true' and sort:
-        return_error('artifact and sort are not supported for scan and scroll together, Please Remove one.')
+        return_error('Please remove or disable one of sort or artifact, As they are not supported in the api together.')
     elif sort and not order:
         return_error('Please specify the order of sorting (Ascending or Descending).')
     elif order and not sort:
         return_error('Please specify a field to sort by.')
     return sort and order
```
Comment (bakatzir) [inline Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py:321]:
```suggestion
    return True
```

File: Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py [hunk 8]
```diff
@@ -297,17 +299,45 @@ def http_request(url_suffix, method='POST', data={}, err_operation=None):
...
+    Args:
+        sort: variable to sort by.
+        order: the order which the results is ordered by.
+        artifact_source: true if artifacts are needed and false otherwise.
+    Returns:
+        true if arguments are valid for the request, false otherwise.
+    """
     if artifact_source == 'true' and sort:
-        return_error('artifact and sort are not supported for scan and scroll together, Please Remove one.')
+        return_error('Please remove or disable one of sort or artifact, As they are not supported in the api together.')
```
Comment (bakatzir) [inline Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py:316]: let's raise an exception

File: Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py [hunk 9]
```diff
@@ -297,17 +299,45 @@ def http_request(url_suffix, method='POST', data={}, err_operation=None):
...
     elif sort and not order:
         return_error('Please specify the order of sorting (Ascending or Descending).')
```
Comment (bakatzir) [inline Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py:318]: let's raise an exception as there is try/catch for the functions

File: Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py [hunk 10]
```diff
@@ -297,17 +299,45 @@ def http_request(url_suffix, method='POST', data={}, err_operation=None):
...
     elif order and not sort:
         return_error('Please specify a field to sort by.')
```
Comment (bakatzir) [inline Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py:320]: let's raise an exception as there is try/catch for the functions

**Comment by orhovy** [comment]
image

**Comment by kirbles19** [comment]
@orhovy Done.

**Comment by bakatzir** [comment]
@Arsenikr is there an eta when we will get access?
if not, please open an issue and yana will remove these tasks from the test playbboks.
@roysagi FYI

**Comment by Arsenikr** [comment]
> @Arsenikr is there an eta when we will get access?
> if not, please open an issue and yana will remove these tasks from the test playbboks.
> @roysagi FYI

Hi, It is a bug on AF side. they said they will fix this in their next release... no ETA thouhg

**Comment by bakatzir** [comment]
@orhovy Please delete these tasks from the TPB.

File: Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py [hunk 11]
```diff
@@ -295,40 +299,91 @@ def http_request(url_suffix, method='POST', data={}, err_operation=None):
     return parse_response(res, err_operation)
 
 
-def validate_sort_and_order(sort, order):
-    if sort and not order:
-        return_error('Please specify the order of sorting (Ascending or Descending).')
-    if order and not sort:
-        return_error('Please specify a field to sort by.')
-    return sort and order
+def validate_sort_and_order_and_artifact(sort: Optional[str], order: Optional[str], artifact_source: Optional[str])\
```
Comment (Itay4) [inline Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py:302]: `Optional` type hint should usually come with `None` as default value

File: Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.yml [hunk 12]
```diff
@@ -385,6 +416,9 @@ script:
     - contextPath: File.Tags
       description: Tags of the file.
       type: String
+    - contextPath: AutoFocus.SessionsResults.Seen
+      description: Session seen.
+      type: Unknown
```
Comment (Itay4) [inline Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.yml:421]: why is this `Unknown`?

File: Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py [hunk 13]
```diff
@@ -295,40 +299,91 @@ def http_request(url_suffix, method='POST', data={}, err_operation=None):
...
+        search_object: Type of search sessions or samples.
+        query: Query based on conditions specified within this object.
+        scope:  Scope of the search. Only available and required for: samples. e.g. Public, Global, Private.
+        size: Number of results to provide.
+        sort: Sort based on the provided artifact.
+        order: How to display sort results in ascending or descending order.
+        artifact_source: Whether artifacts are wanted or not.
+    Returns:
+        dict of response for result commands.
+
```
Comment (Itay4) [inline Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py:376]: remove empty lines

File: Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py [hunk 14]
```diff
@@ -1140,23 +1197,52 @@ def samples_search_results_command():
...
+                return_outputs(readable_output=hr, outputs=context, raw_response=results)
+            else:
+                hr = tableToMarkdown(f'Search Samples Result is {status}', result)
+                hr += tableToMarkdown(f'Artifacts for Sample: ', [])
+                return_outputs(readable_output=hr, outputs=context, raw_response=results)
+
+
+def samples_search_result_hr(result: dict, status: str) -> str:
+    """
+    Function which created a human readable for a specific entry which contains two tables, one for the result's
```
Comment (Itay4) [inline Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.py:1225]:
```suggestion
    Creates human readable output for a specific entry which contains two tables, one for the result's
```

File: Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.yml [hunk 15]
```diff
@@ -134,10 +134,22 @@ script:
       name: last_updated
       required: false
       secret: false
+    - auto: PREDEFINED
+      default: true
```
[the remark this answers is not in the document]
Comment (orhovy) [reply Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.yml:138]: why do you suggest to remove the default value?

[the remark this answers is not in the document]
Comment (Itay4) [reply Packs/AutoFocus/Integrations/AutofocusV2/AutofocusV2.yml:138]: it is not the `defaultValue`, that one is still there
this boolean means that this argument will be `default` in demisto terms

**Comment by Itay4** [comment]
@orhovy i see there is an used function `get_indicator_outputs`
please see if this is a bug and it should be used, or can be removed from the code

**Comment by orhovy** [comment]
@Itay4 I checked and there is no use of this function, I removed it

**Comment by Itay4** [comment]
@orhovy @bakatzir i didnt review the whole thing and just commented a few things i saw by briefly going over
if you're good with it go ahead and merge

**Pull request merged**